java/core/src/test/java/co/worklytics/psoxy/storage/impl/BulkDataSanitizerImplTest.java

```diff
@@ -417,4 +417,54 @@ void pre_v0_4_30_bulk_pseudonym_URL_SAFE_TOKEN_ENCODING(String identifier) {
         assertEquals("t~" + legacyEncoded + (pseudonymizedIdentity.getDomain() == null ? "" : ("@" + pseudonymizedIdentity.getDomain())),
             urlSafeTokenPseudonymEncoder.encode(pseudonymizedIdentity.asPseudonym()));
     }
+
+
+    @SneakyThrows
+    @Test
+    void transform_ghusername() {
+
+        final String EXPECTED = "EMPLOYEE_ID,EMPLOYEE_EMAIL,DEPARTMENT,SNAPSHOT,MANAGER_ID,JOIN_DATE,LEAVE_DATE,GITHUB_USERNAME\r\n" +
+            "2,bob@workltyics.co,Sales,2023-01-06,1,2020-01-01,,\r\n" +
+            "1,alice@worklytics.co,Engineering,2023-01-06,,2019-11-11,,\"{\"\"scope\"\":\"\"github\"\",\"\"hash\"\":\"\"kwv9cWxo7TDgrt1qCegIJv7rA84s_895L_wG_y8hYjA\"\"}\"\r\n" +
+            "4,,Engineering,2023-01-06,1,2018-06-03,,\r\n" +
+            "3,charles@workltycis.co,Engineering,2023-01-06,1,2019-10-06,2022-12-08,\r\n";
+
+        CsvRules rules = CsvRules.builder()
+            .columnsToDuplicate(Map.of("EMPLOYEE_EMAIL", "GITHUB_USERNAME"))
+            .columnsToTransform(Map.of("GITHUB_USERNAME", ColumnarRules.FieldValueTransform.builder()
+                                                                        .filterRegex("(.*)@worklytics.co")
+                                                                        .outputTemplate("%s_enterprise")
```

Comment:
fine for a first approach, but could use named groups and apply replace directly
`(?<username>.*)@worklytics.co`
template `"${username}_enterprise"`, and that will apply directly `replaceAll(template)`
or using numerical capture groups rather than finding "%s" in the code